source/fe/fe_update_flags.cc

```diff
@@ -0,0 +1,140 @@
+  internal::make_update_flags(0);
+constexpr UpdateFlags UpdateFlags::update_values =
+  internal::make_update_flags(0x0001);
+constexpr UpdateFlags UpdateFlags::update_gradients =
+  internal::make_update_flags(0x0002);
+constexpr UpdateFlags UpdateFlags::update_hessians =
+  internal::make_update_flags(0x0004);
+constexpr UpdateFlags UpdateFlags::update_3rd_derivatives =
+  internal::make_update_flags(0x0008);
+constexpr UpdateFlags UpdateFlags::update_boundary_forms =
+  internal::make_update_flags(0x0010);
+constexpr UpdateFlags UpdateFlags::update_quadrature_points =
+  internal::make_update_flags(0x0020);
+constexpr UpdateFlags UpdateFlags::update_JxW_values =
+  internal::make_update_flags(0x0040);
+constexpr UpdateFlags UpdateFlags::update_normal_vectors =
+  internal::make_update_flags(0x0080);
+constexpr UpdateFlags UpdateFlags::update_jacobians =
+  internal::make_update_flags(0x0100);
+constexpr UpdateFlags UpdateFlags::update_jacobian_grads =
+  internal::make_update_flags(0x0200);
+constexpr UpdateFlags UpdateFlags::update_inverse_jacobians =
+  internal::make_update_flags(0x0400);
+constexpr UpdateFlags UpdateFlags::update_covariant_transformation =
+  internal::make_update_flags(0x0800);
+constexpr UpdateFlags UpdateFlags::update_contravariant_transformation =
+  internal::make_update_flags(0x1000);
+constexpr UpdateFlags UpdateFlags::update_transformation_values =
+  internal::make_update_flags(0x2000);
+constexpr UpdateFlags UpdateFlags::update_transformation_gradients =
+  internal::make_update_flags(0x4000);
+constexpr UpdateFlags UpdateFlags::update_volume_elements =
+  internal::make_update_flags(0x10000);
+constexpr UpdateFlags UpdateFlags::update_jacobian_pushed_forward_grads =
+  internal::make_update_flags(0x100000);
+constexpr UpdateFlags UpdateFlags::update_jacobian_2nd_derivatives =
+  internal::make_update_flags(0x200000);
+constexpr UpdateFlags
+  UpdateFlags::update_jacobian_pushed_forward_2nd_derivatives =
+    internal::make_update_flags(0x400000);
+constexpr UpdateFlags UpdateFlags::update_jacobian_3rd_derivatives =
+  internal::make_update_flags(0x800000);
+constexpr UpdateFlags
+  UpdateFlags::update_jacobian_pushed_forward_3rd_derivatives =
+    internal::make_update_flags(0x1000000);
+constexpr UpdateFlags UpdateFlags::update_piola =
+  update_volume_elements | update_contravariant_transformation;
+constexpr UpdateFlags UpdateFlags::update_mapping =
+  update_quadrature_points | update_JxW_values | update_jacobians |
+  update_jacobian_grads | update_jacobian_pushed_forward_grads |
+  update_jacobian_2nd_derivatives |
+  update_jacobian_pushed_forward_2nd_derivatives |
+  update_jacobian_3rd_derivatives |
+  update_jacobian_pushed_forward_3rd_derivatives | update_inverse_jacobians |
+  update_boundary_forms | update_normal_vectors |
+  update_covariant_transformation | update_contravariant_transformation |
+  update_transformation_values | update_transformation_gradients |
+  update_volume_elements;
```

Comment:
What is the difference between this code and lines 21-78 above?

Reply:
It should have been the definition of the global variables. Fixed now.

Reply:
I think I still see the code twice, or which commit is it exactly?